From a11a3b8ce9943951f76c88ab0f595412570547dc Mon Sep 17 00:00:00 2001
From: Mike Kasick <mike@kasick.org>
Date: Mon, 24 Feb 2025 23:32:40 -0500
Subject: [PATCH] meson.build: Support building for knulli.

- Add SDL2_mixer optional dependency for portmaster.
- Install assets, shaders, and license in datadir.
---
 meson.build | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 3555c17..68b0f07 100644
--- a/meson.build
+++ b/meson.build
@@ -221,7 +221,10 @@ if host_machine.system() != 'gba' and host_machine.system() != 'n3ds'
                 sdl_mixer = dependency('sdl2_mixer', static : true, default_options : ['default_library=static'], fallback : ['sdl2_mixer', 'sdl2_mixer_dep'])
             else
                 sdl2_dep = dependency('sdl2', static : false, default_options : ['default_library=shared', 'test=false'], fallback : ['sdl2', 'sdl2_dep'])
-                sdl_mixer = dependency('sdl2_mixer', static : false, default_options : ['default_library=shared'], fallback : ['sdl2_mixer', 'sdl2_mixer_dep'])
+                sdl_mixer = dependency('SDL2_mixer', static : false, required : false)
+                if not sdl_mixer.found()
+                    sdl_mixer = dependency('sdl2_mixer', static : false, default_options : ['default_library=shared'], fallback : ['sdl2_mixer', 'sdl2_mixer_dep'])
+                endif
             endif
         endif
         soloud = dependency('SoLoud', static : true, default_options : ['default_library=static'], fallback : ['SoLoud', 'soloud_dep'])
@@ -372,9 +375,9 @@ if host_machine.system() != 'gba' and host_machine.system() != 'n3ds'
             install_subdir(audio_dir, install_dir : 'Contents/Resources')
             install_subdir('license', install_dir : 'Contents/Resources')
         else
-            install_subdir(audio_dir, install_dir : '.')
-            install_subdir(shaders_dir, install_dir : '.')
-            install_subdir('license', install_dir : '.')
+            install_subdir(audio_dir, install_dir : get_option('datadir'), exclude_directories : ['shaders'])
+            install_subdir(shaders_dir, install_dir : get_option('datadir'))
+            install_subdir('license', install_dir : get_option('datadir'))
         endif
     endif
 elif host_machine.system() == 'n3ds'
-- 
2.43.0

